src/layers/tile-layer-utils/TileCache.js

```diff
@@ -126,4 +147,72 @@ export default class TileCache {
   _getTileId(x, y, z) {
     return `${z}-${x}-${y}`;
   }
+
+  _expandIndices(tileIndices){
+    var [minX, minY, maxY, maxX] = [Number.POSITIVE_INFINITY, Number.POSITIVE_INFINITY, Number.NEGATIVE_INFINITY, Number.NEGATIVE_INFINITY]
+    for (var tileIndex in tileIndices) {
+      var tile = tileIndices[tileIndex]
+      if(tile.x > maxX)
+        maxX = tile.x
+      if(tile.y > maxY)
+        maxY = tile.y
+      if(tile.y < minY)
+        minY = tile.y
+      if(tile.x < minX)
+        minX = tile.x
+    }
```

Comment:
Stepping back, could we assume we'll have a rectangular set of tiles? And then add the adjoining surrounding tiles... or is something more going on here?

Reply:
No that's all that is going on here.  Are you proposing a different way of doing that